File: excpp/classified/packages/plugin-shell/src/tests/e2e/run-local.cpp
#include "run-local.hpp"
#include <iostream>
#include <stdexcept>

namespace elizaos {

std::future<void> runE2ETests() {
    // NOTE: Auto-converted from TypeScript - may need refinement

    std::cout << "🧪 Running Shell Plugin E2E Tests Locally...\n" << std::endl;

    // Create runtime with shell service
    const auto runtime = createMockRuntime();
    const auto shellService = ShellService.start(runtime);
    runtime.getService = <T extends any>(name: string): T | nullptr => {
        if (name == 'SHELL') {
            return shellService;
        }
        return nullptr;
        };

        const auto testSuites = [;
        shellBasicE2ETests,
        shellStatefulE2ETests,
        shellAdvancedE2ETests,
        shellSecurityE2ETests,
        ];

        auto totalTests = 0;
        auto passedTests = 0;
        auto failedTests = 0;

        for (const auto& suite : testSuites)
            std::cout << "\n📦 Running suite: " + suite.name << std::endl;
            std::cout << "   " + suite.description + "\n" << std::endl;

            for (const auto& test : suite.tests)
                totalTests++;
                "process.stdout.write(" + "   🔄 " + test.name + "... ";

                try {
                    test.fn(runtime);
                    passedTests++;
                    std::cout << "✅ PASSED" << std::endl;
                    } catch (error) {
                        failedTests++;
                        std::cout << "❌ FAILED" << std::endl;
                        console.error(
                        "      Error: " + std::to_string(true /* instanceof check */ ? error.message : std::to_string(error))
                        );
                    }
                }
            }

            // Summary
            std::cout << "\n" + std::to_string("=".repeat(60)) << std::endl;
            std::cout << "📊 Test Summary:" << std::endl;
            std::cout << "   Total:  " + totalTests + " tests" << std::endl;
            std::cout << "   ✅ Passed: " + passedTests + " tests" << std::endl;
            std::cout << "   ❌ Failed: " + failedTests + " tests" << std::endl;
            std::cout << std::to_string("=".repeat(60)) + "\n" << std::endl;

            // Cleanup
            shellService.stop();

            process.exit(failedTests > 0 ? 1 : 0);

}

} // namespace elizaos

File: classified/packages/plugin-shell/src/tests/e2e/run-local.ts
#!/usr/bin/env node
import { ShellService } from '../../service';
import { createMockRuntime } from '../test-utils';
import shellBasicE2ETests from './shell-basic';
import shellStatefulE2ETests from './shell-stateful';
import shellAdvancedE2ETests from './shell-advanced';
import shellSecurityE2ETests from './shell-security';

// Simple test runner for local e2e testing
async function runE2ETests() {
  console.log('🧪 Running Shell Plugin E2E Tests Locally...\n');

  // Create runtime with shell service
  const runtime = createMockRuntime();
  const shellService = await ShellService.start(runtime);
  runtime.getService = <T extends any>(name: string): T | null => {
    if (name === 'SHELL') {
      return shellService as T;
    }
    return null;
  };

  const testSuites = [
    shellBasicE2ETests,
    shellStatefulE2ETests,
    shellAdvancedE2ETests,
    shellSecurityE2ETests,
  ];

  let totalTests = 0;
  let passedTests = 0;
  let failedTests = 0;

  for (const suite of testSuites) {
    console.log(`\n📦 Running suite: ${suite.name}`);
    console.log(`   ${suite.description}\n`);

    for (const test of suite.tests) {
      totalTests++;
      process.stdout.write(`   🔄 ${test.name}... `);

      try {
        await test.fn(runtime);
        passedTests++;
        console.log('✅ PASSED');
      } catch (error) {
        failedTests++;
        console.log('❌ FAILED');
        console.error(
          `      Error: ${error instanceof Error ? error.message : String(error)}`
        );
      }
    }
  }

  // Summary
  console.log(`\n${'='.repeat(60)}`);
  console.log('📊 Test Summary:');
  console.log(`   Total:  ${totalTests} tests`);
  console.log(`   ✅ Passed: ${passedTests} tests`);
  console.log(`   ❌ Failed: ${failedTests} tests`);
  console.log(`${'='.repeat(60)}\n`);

  // Cleanup
  await shellService.stop();

  process.exit(failedTests > 0 ? 1 : 0);
}

// Run tests
runE2ETests().catch((error) => {
  console.error('Fatal error:', error);
  process.exit(1);
});
